spec/food_order_spec.rb

```diff
@@ -0,0 +1,56 @@
+require "food_order"
+require "menu"
+require "text_client"
+
+describe FoodOrder do
+
+  let(:dish_list) { { "test dish1" => 6.00, "test dish2" => 4.00, "test dish3" => 4.00 } } 
+  let(:menu_double) { instance_double(Menu, menu: dish_list) }
+
+  let(:formatted_time) { (Time.now + 3600).strftime("%k:%M") }
+  let(:formatted_confirmation) { "Thank you! Your order was placed and will be delivered before #{formatted_time}" }
+
+  let(:text_client_double) { instance_double(TextClient) }
+
+  subject { described_class.new(menu_double, text_client_double) }
+
+  context "placing an order" do
+    before(:example) do
+      subject.add_dish("test dish1")
+      subject.add_dish("test dish3")
+    end
+
+    describe "#add_dish" do
+      it "adds dishes to the customer's basket" do
+        expect(subject.basket).to eq [{ "test dish1" => 6.00 }, { "test dish3" => 4.00 }]
```

Comment:
Should the user have direct access to the basket instance variable? Or should they have access to a method that prints the current basket? In the first case, that fixes the implementation into only one way. In the second, your tests could be based on the behaviour or printing a specific string, and it leaves you with more flexibility on how to change the implementation after.